coreui/src/main/java/com/alexafit/coreui/components/textfield/TextFieldWithIcon.kt

```diff
@@ -17,30 +22,37 @@ import androidx.compose.ui.draw.clip
 import androidx.compose.ui.draw.shadow
 import androidx.compose.ui.focus.FocusState
 import androidx.compose.ui.focus.onFocusChanged
-import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.input.ImeAction
 import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.text.style.TextAlign
 import com.alexafit.coreui.LocalSpacing
 import com.alexafit.coreui.R
-import com.alexafit.coreui.components.buttons.IconActionButton
 
 @Composable
-fun TextField(
+fun TextFieldWithIcon(
     text: String,
     hint: String,
     keyboardImeAction: ImeAction,
-    iconVector: ImageVector?,
-    iconVectorDescription: Int?,
     modifier: Modifier = Modifier,
-    isHintVisible: Boolean = false,
-    isIconClickable: Boolean = false,
-    onValueChange: (String) -> Unit,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    iconImage: Int? = null,
+    iconContentDescription: Int? = null,
+    iconTint: Color = MaterialTheme.colors.onSurface,
```

Comment:
Why not make this a composable instead for maximum flexibility and less parameters?
```
icon: @Composable () -> Unit
```
That's how it's done everywhere else, for example the `Scaffold` composable has a `floatingActionButton: @Composable () -> Unit` lambda